### ISSUE TITLE
Add constructors XmlSerializerNamespaces

### ISSUE THREAD

**Issue by kronic** [dotnet/runtime#41431]
## Background and Motivation
reduction of allocations
## Proposed API
```diff
namespace System.Xml.Serialization
{
+     public XmlSerializerNamespaces(int capacity) { }
+     public XmlSerializerNamespaces(IDictionary<string, string> namespaces) { }
}
```
## Usage Examples
``` C#
public static XmlSerializerNamespaces ToXmlSerializerNamespaces(this IXmlNamespaceResolver xmlNamespaceResolver)
{
    IDictionary<string, string> namespaces = xmlNamespaceResolver.GetNamespacesInScope(XmlNamespaceScope.ExcludeXml);
    return new XmlSerializerNamespaces(namespaces );
}
```
## Alternative Designs
No
## Risks
No

**Comment by KalleOlaviNiemitalo** [comment]
Do you mean `XmlSerializerNamespaces` would save the `IDictionary<string, string> namespaces` reference, instead of copying the keys and values to its private dictionary and calling `ExtensionMethods.ToUri`, which also allocates?

Or would the savings come only from being able to allocate the private dictionary with the correct size, instead of expanding it repeatedly? In which case, an `XmlSerializerNamespaces(int capacity)` constructor would suffice and perhaps be usable in more situations.

**Comment by kronic** [comment]
@KalleOlaviNiemitalo The savings come only from being able to allocate the private dictionary with the correct size, instead of expanding it repeatedly.
Constructor **XmlSerializerNamespaces(IDictionary<string, string> namespaces)** added for convenience.

**Comment by StephenMolloy** [comment]
After discussion with the team, this bug does not seem to merit
addition of new public API's. But there were some small areas for
improvement of allocation and conversion between different
collection types that we addressed in #56438.
